conans/client/cmd/test.py

```diff
@@ -24,18 +24,22 @@ def install_build_and_test(app, conanfile_abs_path, reference, graph_info,
         # Don't use "test_package/build/HASH/" as the build_f
         delete_after_build = False
         test_build_folder = base_folder
+        # Don't mess the test_package folder with the conaninfo.txt and others
+        # We don't have the conanfile loaded yet so we cannot know the generators_folder
+        install_folder = os.path.join(base_folder, "install")
```

Comment:
But this is changing the default install location for all cases, unconditionally, so this might break something? Sounds a bit risky

Reply:
Well, only when the layout is declared, but, please, the title said: `WIP BROKEN` and I removed the reviewers... :P